src/deploy-cromwell-on-azure/Deployer.cs

```diff
@@ -1217,29 +1247,52 @@ private async Task ValidateSubscriptionAndResourceGroupAsync(string subscription
                 .WithLogLevel(HttpLoggingDelegatingHandler.Level.Basic)
                 .Authenticate(azureCredentials);
 
-            var subscriptionExists = (await azure.Subscriptions.ListAsync()).Any(sub => sub.SubscriptionId.Equals(subscriptionId, StringComparison.OrdinalIgnoreCase));
+            var subscriptionExists = (await azure.Subscriptions.ListAsync()).Any(sub => sub.SubscriptionId.Equals(configuration.SubscriptionId, StringComparison.OrdinalIgnoreCase));
 
             if (!subscriptionExists)
             {
-                throw new ValidationException($"Invalid or inaccessible subcription id '{subscriptionId}'. Make sure that subscription exists and that you are either an Owner or have Contributor and User Access Administrator roles on the subscription.", displayExample: false);
+                throw new ValidationException($"Invalid or inaccessible subcription id '{configuration.SubscriptionId}'. Make sure that subscription exists and that you are either an Owner or have Contributor and User Access Administrator roles on the subscription.", displayExample: false);
             }
 
-            if (isUpdate && string.IsNullOrEmpty(resourceGroupName))
+            if (configuration.Update && string.IsNullOrEmpty(configuration.ResourceGroupName))
             {
                 throw new ValidationException($"ResourceGroupName is required for the update.", displayExample: false);
             }
 
-            var rgExists = !string.IsNullOrEmpty(resourceGroupName) && await azureClient.ResourceGroups.ContainAsync(resourceGroupName);
+            var rgExists = !string.IsNullOrEmpty(configuration.ResourceGroupName) && await azureClient.ResourceGroups.ContainAsync(configuration.ResourceGroupName);
 
-            if (!string.IsNullOrEmpty(resourceGroupName) && !rgExists)
+            if (!string.IsNullOrEmpty(configuration.ResourceGroupName) && !rgExists)
             {
                 throw new ValidationException($"If ResourceGroupName is provided, the resource group must already exist.", displayExample: false);
             }
 
+            if (configuration.StorageAccountName != null)
```

Comment:
I'd do this:
```
            if (!configuration.Update && configuration.StorageAccountName != null )
            {
                var existingStorageAccount = await GetExistingStorageAccountAsync(configuration.StorageAccountName);
                
                if (existingStorageAccount == null)
                {
                    throw new ValidationException($"If StorageAccountName is provided, the storage account must already exist and be accessible to the user.", displayExample: false);
                }

                if (existingStorageAccount.RegionName != configuration.RegionName)
                {
                    throw new ValidationException($"If StorageAccountName is provided, it must be in the same region ({configuration.RegionName}).", displayExample: false);
                }
            }

            if (!configuration.Update && configuration.BatchAccountName != null)
            {
                var existingBatchAccount = await GetExistingBatchAccountAsync(configuration.BatchAccountName);

                if (existingBatchAccount == null)
                {
                    throw new ValidationException($"If BatchAccountName is provided, the batch account must already exist, be accessible to the user, and be in the same region ({configuration.RegionName}).", displayExample: false);
                }
            }
```

Reply:
I was presuming that retrieving the StorageAccount and BatchAccount objects was more expensive than checking for name availability.  If I do it the way suggested, I would want to save those objects for later, rather than needing to get those objects at a later time, which is what the code does now.

Reply:
In my fix, I do use GetExisting(Batch/Storage)AccountAsync.  Then I use the results of these later when trying to decide whether to create new accounts.